### ISSUE TITLE
Display SuperCharger loading screen

### ISSUE THREAD

**Issue by thrust26** [stella-emu/stella#286]
Instead of current the simple loading bar, we could emulate the SC BIOS (based on the BIOS ROM).

**Comment by sa666666** [comment]
Maybe not completely related to this, but if we need more proper emulation of a cassette drive, then that would benefit KidVid and Compumate controllers too.  There are already issues created for each of those.

**Comment by Keatah** [comment]
I got thinking about this the other day. Love to see it happen, for realism!

**Comment by sa666666** [comment]
@thrust26, I have a branch with working Supercharger load from MP3/WAV and using the original Supercharger BIOS.  Maybe when we find time, we can address this issue there too.  I'm not merging the branch until I'm 100% sure everything is working.

**Comment by sa666666** [comment]
The code is currently in another branch: `feature/supercharger`.

For anyone wanting to do testing, you need two things:
 - Real supercharger BIOS; name it 'supercharger_bios.bin', and place it in basedir (~/.config/stella for Linux, %APPDATA%/Stella for Windows)
 - Get some MP3/WAV AR files, perhaps from Stella Gets a New Brain! or elsewhere on AtariAge.

I've tested all MP3, but not WAV.  But I suppose one could just convert an MP3 to WAV and test from there.

**Comment by sa666666** [comment]
I've fixed all remaining bugs, and this branch is now merged to master.

**Comment by ale-79** [comment]
> But I suppose one could just convert an MP3 to WAV and test from there.


Attached is the old "Makewav 4.2" utility by Bob Colbert. 

[makewav4.2.zip](https://github.com/user-attachments/files/28908290/makewav4.2.zip)

(To compile in my Linux system, I need to add options  "-Wno-implicit-function-declaration" and "-Wno-implicit-int" to gcc in the Makefile)

You can use it to convert any Supercharger 8448 bytes binary into WAV. Use the option "-ts" (else it generates files for the Cuttle Cart").

**Comment by thrust26** [comment]
> * Real supercharger BIOS; name it 'supercharger_bios.bin', and place it in basedir (~/.config/stella for Linux, %APPDATA%/Stella for Windows)

1. Shouldn't that be %APPDATA%/Roaming/Stella (and in the doc too)? At least that's where it works for me.
2. Also, there is a PAL bios, which should be used for PAL games.
3. All info should be added to the User's Guide

> * Get some MP3/WAV AR files, perhaps from Stella Gets a New Brain! or elsewhere on AtariAge.

I tested _Escape from the Mindmaster_ and get Rewind Tape messages in loads 2..4. I suppose that's OK, because the game knows if the load is allowed, right? How can I cheat?

> I've tested all MP3, but not WAV. But I suppose one could just convert an MP3 to WAV and test from there.

I tested some WAVs for 4K ROMs from the PlusStore (heaps of them there). All good. Did not see anything larger than 4K, though _makewav_ can handle them. I suppose nobody has created any yet (and done the required fixes).

And I noticed that my debug version runs at less than 30Hz.

**Comment by sa666666** [comment]
@thrust26: Yes, those paths were from memory, so I got part of it wrong.  And it does need to be added to the docs, with some snapshots as well.  That's what I asked you about in the other discussion 😈 

Regarding the 30Hz, it's not Stella slowing down, or CPU bound.  This seems to be the rate at which the BIOS processes the audio data.  Since I've never seen a real Supercharger in action, I can't say whether that's accurate or not.

**Comment by thrust26** [comment]
> [@thrust26](https://github.com/thrust26): Yes, those paths were from memory, so I got part of it wrong. And it does need to be added to the docs, with some snapshots as well. That's what I asked you about in the other discussion 😈

I will correct the paths. 

> Regarding the 30Hz, it's not Stella slowing down, or CPU bound. This seems to be the rate at which the BIOS processes the audio data. Since I've never seen a real Supercharger in action, I can't say whether that's accurate or not.

Ah, the frames are not valid frames, but take ~2 twice as long as they should. I suppose digital devices will freak out. 😆
